src/librustc_mir/hair/mod.rs

```diff
@@ -213,10 +213,7 @@ pub enum ExprKind<'tcx> {
     },
     Repeat {
         value: ExprRef<'tcx>,
-        // FIXME(#29789): Add a separate hair::Constant<'tcx> so this could be more explicit about
-        // its contained data. Currently this should only contain expression of ExprKind::Literal
-        // kind.
-        count: ExprRef<'tcx>,
+        count: Constant<'tcx>,
```

Comment:
Then after all, maybe it still make sense to add a new `hair::Constant` struct which contains the value as u64 as suggested by @oli-obk and the span (we could also store the type of the constant there of course)